clusterloader2/pkg/prometheus/prometheus.go

```diff
@@ -342,17 +350,21 @@ func (pc *Controller) GetFramework() *framework.Framework {
 	return pc.framework
 }
 
-func (pc *Controller) applyManifests(manifestGlob string) error {
+func (pc *Controller) applyDefaultManifests(manifestGlob string) error {
+	return pc.applyManifests(pc.clusterLoaderConfig.PrometheusConfig.ManifestPath, manifestGlob)
+}
+
+func (pc *Controller) applyManifests(path, manifestGlob string) error {
 	return pc.framework.ApplyTemplatedManifests(
-		pc.manifestsFS(), manifestGlob, pc.templateMapping, client.Retry(apierrs.IsNotFound))
+		pc.manifestsFS(path), manifestGlob, pc.templateMapping, client.Retry(apierrs.IsNotFound))
 }
 
-func (pc *Controller) manifestsFS() fs.FS {
-	if pc.clusterLoaderConfig.PrometheusConfig.ManifestPath != "" {
-		return os.DirFS(pc.clusterLoaderConfig.PrometheusConfig.ManifestPath)
+func (pc *Controller) manifestsFS(path string) fs.FS {
```

Comment:
I feel like it would be more readable to keep it the way it is and in applyDefaultManifests just use pc.ManifestsFS() and in applyManifests jus use the path passed directly. Then in line 296 you could just run pc.ApplyManifests with os.DirFS(additionalPath)

The problem that I have is that when reading the code I had to remind myself that ManifestPath in prometheus config is deprecated and we want to default to embedded . But in the case of additionalMonitorsPath we do not want to default to embedded manifests and you need to do additional check in line 294.

Reply:
ah, I missed it's actually deprecated. How about I just remove it in that case?

Reply:
👍  I've checked and it does not seem to be used in perf-tests/test-infra repositories.

Reply:
pushed new commit with removal, PTAL